src/System.Runtime/tests/System/GCTests.cs

```diff
@@ -157,14 +169,19 @@ public static void KeepAlive_Null()
 
         private class KeepAliveNullTest
         {
-            public static void Run()
+            [System.Runtime.CompilerServices.MethodImplAttribute(System.Runtime.CompilerServices.MethodImplOptions.NoInlining)]
+            private static void MakeAndNull()
             {
                 var obj = new TestObject();
                 obj = null;
+            }
+
+            public static void Run()
+            {
+                MakeAndNull();
 
                 GC.Collect();
                 GC.WaitForPendingFinalizers();
-                GC.KeepAlive(obj);
 
```

Comment:
What did the GC.KeepLive(obj) try to achieve before your change? How could the test work at all?